sklearn_numba_dpex/kmeans/kernels/compute_euclidean_distances.py

```diff
@@ -30,28 +27,25 @@ def make_compute_euclidean_distances_fixed_window_kernel(
     )
 
     (
-        _initialize_window_of_centroids,
-        _load_window_of_centroids_and_features,
-    ) = _make_initialize_window_kernel_funcs(
-        n_clusters,
-        n_features,
-        work_group_size,
-        window_n_centroids,
-        centroids_window_height,
-        dtype,
-    )
-
-    _accumulate_sq_distances = _make_accumulate_sum_of_ops_kernel_func(
+        initialize_window_of_centroids,
+        load_window_of_centroids_and_features,
+        accumulate_sq_distances,
+    ) = make_pairwise_ops_base_kernel_funcs(
         n_samples,
         n_features,
+        n_clusters,
         centroids_window_height,
         window_n_centroids,
         ops="squared_diff",
         dtype=dtype,
+        work_group_size=work_group_size,
+        initialize_window_of_centroids_half_l2_norms=False,
     )
 
     n_windows_for_centroids = math.ceil(n_clusters / window_n_centroids)
     n_windows_for_features = math.ceil(n_features / centroids_window_height)
+    last_centroid_window_idx = n_windows_for_centroids - 1
+    last_feature_window_idx = n_windows_for_features - 1
 
     centroids_window_shape = (centroids_window_height, (window_n_centroids + 1))
```

Comment:
Here again, why the "+ 1"?

Reply:
There's some optimization regarding IO in shared memory that consists in adding padding to avoid bank conflicts and I thought that was good here. But in fact I'm not sure it's enough, could remove the +1 and leave a ticket / TODO for later investigation.